Sources/Misc/SystemInfo.swift

```diff
@@ -144,6 +144,11 @@ class SystemInfo {
         self.responseVerificationMode = responseVerificationMode
         self.dangerousSettings = dangerousSettings ?? DangerousSettings()
         self.clock = clock
+        // eager-load the isSandbox value from a worker thread so that's immediately available
+        // this is useful on macOS where it may take long for the value to compute
+        self.operationDispatcher.dispatchOnWorkerThread {
+            _ = self._isSandbox
+        }
```

Comment:
this pre-loading makes the changes quite safe. After all, for the most part we'll only be loading once, as soon as SystemInfo is created, and from a worker thread. 
So other accesses will likely not hit this from a different thread. Ideally we'd modify the property to be asynchronous though so that we're completely covered.

Reply:
Hmm I imagine we could potentially be calculating the `isSandbox` property multiple times at the same time right? One here on initialization in a worker thread, then another in the main thread, if that property gets accessed from the main thread (Or even more if accessed from multiple threads). I think this might be ok, but I wanted to mention it.

I also see that we are accessing the `sandboxEnvironmentDetector.isSandbox` in multiple places, like the `DeviceCache` or the `StoreKit1Wrapper`. Do we need to handle those as well? (That can happen in different PRs though)